miden/src/cli/data.rs

```diff
@@ -404,13 +410,18 @@ impl ProgramFile {
 
     /// Compiles this program file into a [Program].
     #[instrument(name = "compile_program", skip_all)]
-    pub fn compile<'a, I, L>(&self, debug: &Debug, libraries: I) -> Result<Program, Report>
+    pub fn compile<'a, I, L>(
+        &self,
+        debug: &Debug,
+        libraries: I,
+        source_manager: Arc<dyn assembly::SourceManager>,
+    ) -> Result<Program, Report>
     where
         I: IntoIterator<Item = &'a L>,
         L: ?Sized + Library + 'static,
     {
         // compile program
-        let mut assembler = Assembler::default()
+        let mut assembler = Assembler::new(source_manager.clone())
```

Comment:
Yes, because we have callers of `compile` which parse source files and load libraries, which require a `SourceManager` impl, and we need to perform assembly using the same source manager in order for spans to be valid.